tests/unit/test_dataset.py

```diff
@@ -141,18 +141,35 @@ def test_to_api_repr_routine(self):
         exp_resource = {"routine": routine}
         self.assertEqual(resource, exp_resource)
 
+    def test_to_api_repr_dataset(self):
+        dataset = {
+            "dataset": {"projectId": "my-project", "datasetId": "my_dataset"},
+            "target_types": "VIEWS",
+        }
+        entry = self._make_one(None, "dataset", dataset)
+        resource = entry.to_api_repr()
+        exp_resource = {"dataset": dataset}
+        self.assertEqual(resource, exp_resource)
+
+    def test_to_api_w_incorrect_role(self):
+        dataset = {
+            "dataset": {
+                "projectId": "my-project",
+                "datasetId": "my_dataset",
+                "tableId": "my_table",
+            },
+            "target_type": "VIEW",
+        }
+        with self.assertRaises(ValueError):
+            self._make_one("READER", "dataset", dataset)
+
     def test_from_api_repr(self):
         resource = {"role": "OWNER", "userByEmail": "salmon@example.com"}
         entry = self._get_target_class().from_api_repr(resource)
         self.assertEqual(entry.role, "OWNER")
         self.assertEqual(entry.entity_type, "userByEmail")
         self.assertEqual(entry.entity_id, "salmon@example.com")
 
-    def test_from_api_repr_w_unknown_entity_type(self):
```

Comment:
Could we update this to check that it successfully creates an AccessEntry? I'm a bit worried still that unknown types will cause runtime errors if the backend ever adds another access type.

We may want to cache the resource from `from_api_repr` in a `_properties` value as a stepping stone towards https://github.com/googleapis/python-bigquery/issues/1077 even if we don't use it for all the properties (yet).